src/native-stack/views/NativeStackView.tsx

```diff
@@ -260,128 +261,130 @@ const RouteView = ({
-      statusBarColor={statusBarColor}
-      statusBarHidden={statusBarHidden}
-      statusBarStyle={statusBarStyle ?? (dark ? 'light' : 'dark')}
-      statusBarTranslucent={statusBarTranslucent}
-      swipeDirection={swipeDirection}
-      transitionDuration={transitionDuration}
-      onHeaderBackButtonClicked={() => {
-        navigation.dispatch({
-          ...StackActions.pop(),
-          source: route.key,
-          target: stateKey,
-        });
-      }}
-      onWillAppear={() => {
-        navigation.emit({
-          type: 'transitionStart',
-          data: { closing: false },
-          target: route.key,
-        });
-      }}
-      onWillDisappear={() => {
-        navigation.emit({
-          type: 'transitionStart',
-          data: { closing: true },
-          target: route.key,
-        });
-      }}
-      onAppear={() => {
-        navigation.emit({
-          type: 'appear',
-          target: route.key,
-        });
-        navigation.emit({
-          type: 'transitionEnd',
-          data: { closing: false },
-          target: route.key,
-        });
-      }}
-      onDisappear={() => {
-        navigation.emit({
-          type: 'transitionEnd',
-          data: { closing: true },
-          target: route.key,
-        });
-      }}
-      onHeaderHeightChange={e => {
-        const headerHeight = e.nativeEvent.headerHeight;
-
-        if (cachedAnimatedHeaderHeight.current !== headerHeight) {
-          // Currently, we're setting value by Animated#setValue, because we want to cache animated value.
-          // Also, in React Native 0.72 there was a bug on Fabric causing a large delay between the screen transition,
-          // which should not occur.
-          // TODO: Check if it's possible to replace animated#setValue to Animated#event.
-          animatedHeaderHeight.setValue(headerHeight);
-          cachedAnimatedHeaderHeight.current = headerHeight;
-        }
-      }}
-      onDismissed={e => {
-        navigation.emit({
-          type: 'dismiss',
-          target: route.key,
-        });
-
-        const dismissCount =
-          e.nativeEvent.dismissCount > 0 ? e.nativeEvent.dismissCount : 1;
-
-        navigation.dispatch({
-          ...StackActions.pop(dismissCount),
-          source: route.key,
-          target: stateKey,
-        });
-      }}
-      onGestureCancel={() => {
-        navigation.emit({
-          type: 'gestureCancel',
-          target: route.key,
-        });
-      }}>
-      <AnimatedHeaderHeightContext.Provider value={animatedHeaderHeight}>
-        <HeaderHeightContext.Provider value={staticHeaderHeight}>
-          <MaybeNestedStack
-            options={options}
-            route={route}
-            stackPresentation={stackPresentation}>
-            {renderScene()}
-          </MaybeNestedStack>
-          {/* HeaderConfig must not be first child of a Screen.
+    <ScreenInfoContext.Provider value={{ options }}>
+      <Screen
```

Comment:
Wrapping the screen inside context here will work only with navigation v5. v6 (hosted in `react-navigation` repo) uses its own `NativeStackView` & won't be wrapped inside such context, thus I'm postulating following action points:

1. We need to make sure that these changes are ported for native stack v6
2. We need to make sure that there won't be any crashes in case the context is not present

Reply:
Sure! Resolved in https://github.com/software-mansion/react-native-screens/pull/1915/commits/b2d876ac20a3972688451fdf3f9b62c12086f6c2.